lib/active_model/serializer/association.rb

```diff
@@ -9,10 +9,58 @@ class Serializer
     # @example
     #  Association.new(:comments, CommentSummarySerializer)
     #
-    Association = Struct.new(:name, :serializer, :options, :links, :meta) do
+    Association = Struct.new(:name, :serializer, :options) do
+      attr_reader :reflection_options, :object
+
+      def initialize(*)
+        super
+        @reflection_options = options.delete(:new_options)
+      end
+
       # @return [Symbol]
       def key
-        options.fetch(:key, name)
+        @key ||= options.fetch(:key) { name }
+      end
+
+      def links
```

Comment:
So, something that's confusing for me, is that this association here is a struct, where as the generic reflection defined below is a class, and they are both under the `active_model/serializer` namespace.  *I* know they have different purposes, but I think the organization of the classes should reflect the different purposes as well. Maybe a namespace for definitions, and a namespace for representation? or maybe just a namespace for representation, since in a serializer, we're most often working with the definitions of things.

thoughts?